### ISSUE TITLE
Error in selecting --output-spaces MNIPediatric:cohort-5:res-2 

### ISSUE THREAD

**Issue by hfxcarl** [nipreps/fmriprep#3246]
### What happened?

Running latest 23.2.0 fmriprep on QTAB OpenNeuro dataset and specifying --output-spaces MNIPediatric:cohort-5:res-2.  

### What command did you use?

```shell
singularity run --cleanenv \
    -B /OpenData/QTAB/QTAB/BIDS_raw:/data:ro \
    -B /OpenData/QTAB/QTAB/BIDS_raw_DB:/pybids_db:ro \
    -B /OpenData/QTAB/QTAB/derivatives/fmriprep:/derivs \
    -B /scratch/QTAB/QTAB/fmriprep:/work \
    -B /opt/templateflow:/templateflow \
    /opt/SingularityImgs/fmriprep-23.2.0.sif \
    /data \
    /derivs \
    participant \
    --participant_label 0005 \
    -w /work \
    --bids-database-dir /pybids_db \
    --nprocs 8 \
    --omp-nthreads 8 \
    --mem 16  -v \
    --no-sub \
    --resource-monitor \
    --skip_bids_validation \
    --notrack \
    --output-spaces MNIPediatricAsym:res-2:cohort-5 MNI152NLin2009cAsym:res-2 \
    --cifti-output
```

### What version of fMRIPrep are you running?

23.2.0

### How are you running fMRIPrep?

Singularity

### Is your data BIDS valid?

Yes

### Are you reusing any previously computed results?

No

### Please copy and paste any relevant log output.

```shell
## log:
240301-15:16:51,788 nipype.workflow IMPORTANT:
         Running fMRIPrep version 23.2.0

         License NOTICE ##################################################
         fMRIPrep 23.2.0
         Copyright 2023 The NiPreps Developers.
         
         This product includes software developed by
         the NiPreps Community (https://nipreps.org/).
         
         Portions of this software were developed at the Department of
         Psychology at Stanford University, Stanford, CA, US.
         
         This software is also distributed as a Docker container image.
         The bootstrapping file for the image ("Dockerfile") is licensed
         under the MIT License.
         
         This software may be distributed through an add-on package called
         "Docker Wrapper" that is under the BSD 3-clause License.
         #################################################################
240301-15:17:24,665 nipype.workflow IMPORTANT:
         Building fMRIPrep's workflow:
           * BIDS dataset path: /data.
           * Participant list: ['0005'].
           * Run identifier: 20240301-151603_c1ba3932-78fe-429c-b836-29d64fbe294b.
           * Output spaces: MNIPediatricAsym:cohort-5:res-2 MNI152NLin2009cAsym:res-2.
           * Pre-run FreeSurfer's SUBJECTS_DIR: /derivs/sourcedata/freesurfer.
240301-15:17:48,878 nipype.workflow INFO:
         ANAT Stage 1: Adding template workflow
240301-15:17:50,199 nipype.workflow INFO:
         ANAT Stage 2: Preparing brain extraction workflow
240301-15:17:50,262 nipype.workflow INFO:
         ANAT Stage 3: Preparing segmentation workflow
240301-15:17:50,265 nipype.workflow INFO:
         ANAT Stage 4: Preparing normalization workflow for ['MNIPediatricAsym:cohort-5', 'MNI152NLin2009cAsym', 'MNI152NLin6Asym']
240301-15:17:50,285 nipype.workflow INFO:
         ANAT Stage 5: Preparing surface reconstruction workflow
240301-15:17:50,301 nipype.workflow INFO:
         ANAT Stage 6: Preparing mask refinement workflow
240301-15:17:50,303 nipype.workflow INFO:
         ANAT Stage 7: Creating T2w template
240301-15:17:50,309 nipype.workflow INFO:
         ANAT Stage 8: Creating GIFTI surfaces for ['white', 'pial', 'midthickness', 'sphere_reg', 'sphere']
240301-15:17:50,325 nipype.workflow INFO:
         ANAT Stage 8: Creating GIFTI metrics for ['thickness', 'sulc']
240301-15:17:50,331 nipype.workflow INFO:
         ANAT Stage 8a: Creating cortical ribbon mask
240301-15:17:50,334 nipype.workflow INFO:
         ANAT Stage 9: Creating fsLR registration sphere
240301-15:17:50,337 nipype.workflow INFO:
         ANAT Stage 10: Creating MSM-Sulc registration sphere
240301-15:19:21,170 nipype.workflow INFO:
         No single-band-reference found for sub-0005_ses-01_task-rest_dir-AP_bold.nii.gz.
240301-15:19:21,231 nipype.workflow INFO:
         Stage 1: Adding HMC boldref workflow
240301-15:19:21,236 nipype.workflow INFO:
         Stage 2: Adding motion correction workflow
240301-15:19:21,242 nipype.workflow INFO:
         Stage 3: Adding coregistration boldref workflow
240301-15:19:21,272 nipype.workflow IMPORTANT:
         BOLD series will be slice-timing corrected to an offset of 0.416s.
…
…
…
240301-21:04:26,746 nipype.workflow INFO:
         [MultiProc] Running 0 tasks, and 32 jobs ready. Free memory (GB): 0.02/0.02, Free processors: 8/8.
240301-21:04:26,869 nipype.workflow INFO:
         [Node] Setting-up "fmriprep_23_2_wf.sub_0005_wf.select_MNI6" in "/work/fmriprep_23_2_wf/sub_0005_wf/select_MNI6".
240301-21:04:26,871 nipype.workflow INFO:
         [Node] Executing "select_MNI6" <niworkflows.interfaces.utility.KeySelect>
240301-21:04:27,135 nipype.workflow INFO:
         [Node] Finished "select_MNI6", elapsed time 0.000601s.
240301-21:04:27,136 nipype.workflow INFO:
         [Job 453] Completed (fmriprep_23_2_wf.sub_0005_wf.select_MNI6).
240301-21:04:27,259 nipype.workflow INFO:
         [Node] Setting-up "fmriprep_23_2_wf.sub_0005_wf.template_iterator_wf.select_xfm" in "/work/fmriprep_23_2_wf/sub_0005_wf/template_iterator_wf/_in_tuple_MNIPediatricAsymcohort-5.cohort5.res2/select_xfm".
240301-21:04:27,261 nipype.workflow INFO:
         [Node] Executing "select_xfm" <niworkflows.interfaces.utility.KeySelect>
240301-21:04:27,526 nipype.workflow INFO:
         [Node] Finished "select_xfm", elapsed time 0.000435s.
240301-21:04:27,526 nipype.workflow WARNING:
         Storing result file without outputs
240301-21:04:27,526 nipype.workflow WARNING:
         [Node] Error on "fmriprep_23_2_wf.sub_0005_wf.template_iterator_wf.select_xfm" (/work/fmriprep_23_2_wf/sub_0005_wf/template_iterator_wf/_in_tuple_MNIPediatricAsymcohort-5.cohort5.res2/select_xfm)
240301-21:04:27,526 nipype.workflow ERROR:
         Node select_xfm.a0 failed to run on host worf.
240301-21:04:27,527 nipype.workflow ERROR:
         Saving crash info to /derivs/sub-0005/log/20240301-151603_c1ba3932-78fe-429c-b836-29d64fbe294b/crash-20240301-210427-uher-select_xfm.a0-68ceb294-2750-489c-acc8-5ab045166844.txt
Traceback (most recent call last):
  File "/opt/conda/envs/fmriprep/lib/python3.10/site-packages/nipype/pipeline/plugins/multiproc.py", line 344, in _send_procs_to_workers
    self.procs[jobid].run(updatehash=updatehash)
  File "/opt/conda/envs/fmriprep/lib/python3.10/site-packages/nipype/pipeline/engine/nodes.py", line 527, in run
    result = self._run_interface(execute=True)
  File "/opt/conda/envs/fmriprep/lib/python3.10/site-packages/nipype/pipeline/engine/nodes.py", line 645, in _run_interface
    return self._run_command(execute)
  File "/opt/conda/envs/fmriprep/lib/python3.10/site-packages/nipype/pipeline/engine/nodes.py", line 771, in _run_command
    raise NodeExecutionError(msg)
nipype.pipeline.engine.nodes.NodeExecutionError: Exception raised while executing Node select_xfm.

Traceback:
        Traceback (most recent call last):
          File "/opt/conda/envs/fmriprep/lib/python3.10/site-packages/nipype/interfaces/base/core.py", line 400, in run
            outputs = self.aggregate_outputs(runtime)
          File "/opt/conda/envs/fmriprep/lib/python3.10/site-packages/nipype/interfaces/base/core.py", line 429, in aggregate_outputs
            predicted_outputs = self._list_outputs()  # Predictions from _list_outputs
          File "/opt/conda/envs/fmriprep/lib/python3.10/site-packages/niworkflows/interfaces/utility.py", line 217, in _list_outputs
            index = self.inputs.keys.index(self.inputs.key)
        ValueError: 'MNIPediatricAsym:cohort-5' is not in list


240301-21:04:27,546 nipype.workflow INFO:
         [Job 454] Completed (fmriprep_23_2_wf.sub_0005_wf.template_iterator_wf.select_xfm).
240301-21:04:27,672 nipype.workflow INFO:
         [Node] Setting-up "fmriprep_23_2_wf.sub_0005_wf.template_iterator_wf.select_xfm" in "/work/fmriprep_23_2_wf/sub_0005_wf/template_iterator_wf/_in_tuple_MNI152NLin2009cAsym.res2/select_xfm".
240301-21:04:27,674 nipype.workflow INFO:
         [Node] Executing "select_xfm" <niworkflows.interfaces.utility.KeySelect>
240301-21:04:27,941 nipype.workflow INFO:
         [Node] Finished "select_xfm", elapsed time 0.000498s.
240301-21:04:27,942 nipype.workflow INFO:
         [Job 455] Completed (fmriprep_23_2_wf.sub_0005_wf.template_iterator_wf.select_xfm).
```

### Additional information / screenshots

I think there might be a naming issue including "cohort-5" twice?  (_in_tuple_MNIPediatricAsymcohort-5.cohort5.res2):

Specific line from above log:
240301-21:04:27,259 nipype.workflow INFO:
         [Node] Setting-up "fmriprep_23_2_wf.sub_0005_wf.template_iterator_wf.select_xfm" in "/work/fmriprep_23_2_wf/sub_0005_wf/template_iterator_wf/_in_tuple_MNIPediatricAsymcohort-5.cohort5.res2/select_xfm".

**Comment by mgxd** [comment]
This was fixed in https://github.com/nipreps/smriprep/issues/416, hoping to get a new release out with the fix this week.

BTW, I believe the `MNIPediatricAsym:res-2:cohort-5` in your command should be `MNIPediatricAsym:cohort-5:res-2`, as cohort is parsed before resolution.

**Comment by effigies** [comment]
We should probably not be sensitive to the order of the modifiers.

**Comment by WangYunHong98** [comment]
Hi all @effigies @mgxd 

I have the same issue ```ValueError: ‘MNIPediatricAsym:cohort-1’ is not in list``` in fmriprep 23.2.1. 
![b6d047607a2e8de69b8f00416b6912d](https://github.com/nipreps/fmriprep/assets/104638954/1692cc8d-3caf-4334-a3c6-41d17bf5a7a3)
![a8d73a18a600893d4be5e52fa01e003](https://github.com/nipreps/fmriprep/assets/104638954/473ed41c-74ca-49c2-8441-470ffde6ae17)
And this is the argument ```--output-spaces MNIPediatricAsym:res-1:cohort-1``` used in my script.

Has this issue been solved or can I use other versions released before fmriprep 23.2.1.

Best,
Yunhong

**Comment by mgxd** [comment]
@WangYunHong98 sorry for the delay - this is fixed in `24.0.0`

**Comment by WangYunHong98** [comment]
@mgxd It's fine. Thank you for the excellent work you do for fmriprep!
